modules/coredns/collect.go

```diff
@@ -3,30 +3,44 @@ package coredns
 import (
 	"fmt"
 
+	"github.com/blang/semver/v4"
 	"github.com/netdata/go.d.plugin/pkg/prometheus"
 	"github.com/netdata/go.d.plugin/pkg/stm"
 )
 
+const (
+	metricPanicCountTotal169orOlder         = "coredns_panic_count_total"
+	metricRequestCountTotal169orOlder       = "coredns_dns_request_count_total"
+	metricRequestTypeCountTotal169orOlder   = "coredns_dns_request_type_count_total"
+	metricResponseRcodeCountTotal169orOlder = "coredns_dns_response_rcode_count_total"
+
+	metricPanicCountTotal170orNewer         = "coredns_panics_total"
+	metricRequestCountTotal170orNewer       = "coredns_dns_requests_total"
+	metricRequestTypeCountTotal170orNewer   = "coredns_dns_requests_total"
+	metricResponseRcodeCountTotal170orNewer = "coredns_dns_responses_total"
+)
+
 var (
 	empty                  = ""
 	dropped                = "dropped"
 	emptyServerReplaceName = "empty"
 	rootZoneReplaceName    = "root"
+	version169             = semver.MustParse("1.6.9")
 )
 
 var (
-	metricPanicCountTotal = "coredns_panic_count_total"
+	metricPanicCountTotal string
 	// true for all metrics below:
 	// - if none of server block matches 'server' tag is "", empty server has only one zone - dropped.
 	//   example:
-	//   coredns_dns_request_count_total{family="1",proto="udp",server="",zone="dropped"} 1
+	//   coredns_dns_requests_total{family="1",proto="udp",server="",zone="dropped"} 1 for
 	// - dropped requests are added to both dropped and corresponding zone
 	//   example:
-	//   coredns_dns_request_count_total{family="1",proto="udp",server="dns://:53",zone="dropped"} 2
-	//   coredns_dns_request_count_total{family="1",proto="udp",server="dns://:53",zone="ya.ru."} 2
-	metricRequestCountTotal       = "coredns_dns_request_count_total"
-	metricRequestTypeCountTotal   = "coredns_dns_request_type_count_total"
-	metricResponseRcodeCountTotal = "coredns_dns_response_rcode_count_total"
+	//   coredns_dns_requests_total{family="1",proto="udp",server="dns://:53",zone="dropped"} 2
+	//   coredns_dns_requests_total{family="1",proto="udp",server="dns://:53",zone="ya.ru."} 2
+	metricRequestCountTotal       string
+	metricRequestTypeCountTotal   string
+	metricResponseRcodeCountTotal string
```

Comment:
These are global vars and we change them. Won't work if there are 2 jobs: the first collects from v1.6.9, second from v1.7.0+

Reply:
Yes, that's true. I moved this to a struct and make this struct a variable to the CoreDNS object so that each job will have the correct values

Reply:
@ilyam8 Please let me know if that's ok now...

Reply:
Looks good 👍 Thanks!